yt/utilities/tests/test_load_sample.py

```diff
@@ -0,0 +1,35 @@
+from yt.testing import assert_equal
+from yt.utilities.load_sample import _validate_sampledata_name
+
+names = { "t1": {"load_name" : "IsolatedGalaxy.tar.gz",
+                 "answers": { "fileext" : "IsolatedGalaxy.tar.gz",
+                              "basename" : "IsolatedGalaxy",
+                              "extension" : "tar"}},
+          "t2": {"load_name" : "IsolatedGalaxy",
+                 "answers": { "fileext" : "IsolatedGalaxy.tar.gz",
+                              "basename" : "IsolatedGalaxy",
+                              "extension" : "tar"}},
+          "t3": {"load_name" : "apec_emissivity_v3.h5",
+                 "answers": { "fileext" : "apec_emissivity_v3.h5",
+                              "basename" : "apec_emissivity_v3",
+                              "extension" : "h5"}},
+          "t4": {"load_name" : "apec_emissivity_v3.hdf5",
+                 "answers": { "fileext" : "apec_emissivity_v3.hdf5",
+                              "basename" : "apec_emissivity_v3",
+                              "extension" : "h5"}},
+          "t5": {"load_name" : "solution-00027.0000.vtu",
+                 "answers": { "fileext" : "solution-00027.0000.vtu",
+                              "basename" : "solution-00027.0000",
+                              "extension" : ".vtu"}}
+        }
+
+def test_name_validator():
+    for test in names:
+        fileext, bname, ext = _validate_sampledata_name(names[test]["load_name"])
+        expected_answers = names[test]["answers"]
```

Comment:
suggestion:
```python
    for test_key, test_names in names.items():
        fileext, bname, ext = _validate_sampledata_name(test_names["load_name"])
        expected_answers = test_names["answers"]
```

Reply:
I didn't update this because I think both are equally clear.